### PR TITLE
[Testing] [Tooling] Replace godog with gocuke

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Description

Replaces the `godog` gherkin runner dependency with `gocuke`, an alternative which better supports the workflow / developer experience we're looking for. Specifically, it tightly integrates with go's standard `testing.T` type such that we can use third-party libs like `gomock` and `require` which depend on `testing.T`.

## Issue

No issue.

## Type of change

Please mark the relevant option(s):

- [x] New feature, functionality or library
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Replace `godog` with `gocuke`
- Refactor existing step definitions

## Testing

- [ ] `make develop_test`; if any code changes were made
- [ ] [Docker Compose LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md); if any major functionality was changed or introduced
- [ ] [k8s LocalNet](https://github.com/pokt-network/pocket/blob/main/build/localnet/README.md); if any infrastructure or configuration changes were made

## Required Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added, or updated, [`godoc` format comments](https://go.dev/blog/godoc) on touched members (see: [tip.golang.org/doc/comment](https://tip.golang.org/doc/comment))
- [ ] I have tested my changes using the available tooling
- [ ] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [ ] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

**Review by dylanlott — APPROVED** [review]
LGTM ✅ Passes tests on my local. Having require in the E2E test functions will be really nice.

**Comment by bryanchriswhite** [comment]
@dylanlott I'm just now realizing there was no changelog update, yet CI passed. CC: @Olshansk, I don't think this is the first time I've observed this.

**Comment by dylanlott** [comment]
@bryanchriswhite I added the cl-validate label, that should trigger the changelog check.

**Comment by dylanlott** [comment]
Bump to update documentation to reflect any relevant changes in the [E2E Testing Framework](https://github.com/pokt-network/pocket/pull/739).

**Comment by bryanchriswhite** [comment]
Perhaps I didn't understand #691 as well as I thought.

File: shared/k8s/debug.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ func init() {
 	logger.Global.Info().Str("namespace", CurrentNamespace).Msg("got new namespace")
 }
 
+// FetchValidatorPrivateKeys returns a map corresponding to the data section of
+// the validator private keys k8s secret (yaml), located at `privateKeysSecretResourceName`.
```
[the remark this answers is not in the document]
Comment (Olshansk) [reply shared/k8s/debug.go:33]: 💯

File: e2e/tests/steps_init_test.go [hunk 2]
```diff
@@ -45,189 +35,164 @@ const (
...
+	gocuke.TestingT
+
+	// validatorKeys is hydrated by the clientset with credentials for all validators.
+	// validatorKeys maps validator IDs to their private key as a hex string.
+	validatorKeys map[string]string
+	// clientset is the kubernetes API we acquire from the user's $HOME/.kube/config
+	clientset *kubernetes.Clientset
+	// validator holds command results between runs and reports errors to the test suite
+	validator *validatorPod
+	// validatorA maps to suffix ID 001 of the kube pod that we use as our control agent
```
Comment (Olshansk) [inline e2e/tests/steps_init_test.go:48]: Remove this comment?

**Review by Olshansk — COMMENTED** [review]
giphy

**Comment by bryanchriswhite** [comment]
> @dylanlott I'm just now realizing there was no changelog update, yet CI passed. CC: @Olshansk, I don't think this is the first time I've observed this.

This should be resolved in #768

File: e2e/tests/steps_init_test.go (hunk 2 above)
Comment (bryanchriswhite) [reply e2e/tests/steps_init_test.go:48]: Nice catch - thanks! :raised_hands: I'll track this change locally and include it in a forthcoming PR.

**Pull request merged**